src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -895,19 +895,24 @@ public static boolean hasNoElements(Ingredient ingredient) {
     @Deprecated(forRemoval = true, since = "1.20.1") // Tags use a codec now This was never used in 1.20
     public static <T> void deserializeTagAdditions(List<TagEntry> list, JsonObject json, List<TagEntry> allList) {}
 
+    public static final int VANILLA_SERIALIZER_LIMIT = 256;
+
     @Nullable
     public static EntityDataSerializer<?> getSerializer(int id, CrudeIncrementalIntIdentityHashBiMap<EntityDataSerializer<?>> vanilla) {
         EntityDataSerializer<?> serializer = vanilla.byId(id);
         if (serializer == null) {
-            return NeoForgeRegistries.ENTITY_DATA_SERIALIZERS.byId(id);
+            return NeoForgeRegistries.ENTITY_DATA_SERIALIZERS.byId(id - VANILLA_SERIALIZER_LIMIT);
         }
         return serializer;
     }
 
     public static int getSerializerId(EntityDataSerializer<?> serializer, CrudeIncrementalIntIdentityHashBiMap<EntityDataSerializer<?>> vanilla) {
         int id = vanilla.getId(serializer);
         if (id < 0) {
-            return NeoForgeRegistries.ENTITY_DATA_SERIALIZERS.getId(serializer);
+            id = NeoForgeRegistries.ENTITY_DATA_SERIALIZERS.getId(serializer);
+            if (id >= 0) {
+                return id + VANILLA_SERIALIZER_LIMIT;
+            }
```

Comment:
Hmmm. I wonder if there's the potential for accidental uses of the NF registry for (modded) entity data serializer for retrieving the serializer ID, instead of the proper way through `EntityDataSerializers#getSerializedId` (which calls this method). Should we be concerned about that possibility?

Also, a fun note is the tiny mismatch between this method `getSerializerId` and `EntityDataSerializers`' `getSerializedId`. 😄 (Perhaps we ought to fix that?)

Reply:
The registry being used directly to lookup or resolve IDs is indeed a potential issue but I'm not sure introducing any countermeasures against that is particularly feasible and reintroducing the min ID limit on the registry just for this doesn't seem like a particularly great idea either.

`getSerializedId()` feels like a typo to me and a confusing one at that (at least to my mind), not sure adapting our method name makes much sense in that regard

Reply:
Fair enough on both counts.